src/data/markdown/translated-guides/en/06 Testing Guides/04 Running large tests.md

```diff
@@ -5,18 +5,17 @@ excerpt: 'How to run large-scale k6 tests without distributed-execution'
 
 This document explains how to launch a large-scale k6 test on a single machine without the need for distributed execution.
 
-The common misconception of many load testers is that [distributed execution](#distributed-execution) (ability to launch a load test on multiple machines) is required to generate large load. This is not the case with k6.
+A common misconception of many load testers is that [distributed execution](#distributed-execution) (the ability to launch a load test from multiple machines) is required to generate large load. This is not the case with k6.
 
-k6 is different from many other load testing tools in the way it handles hardware resources. A single k6 process will efficiently use all CPU cores on a load generator machine.
-A single instance of k6 is often enough to generate load of 30,000-40,000 simultaneous users (VUs). This number of VUs can generate upwards of 300,000 requests per second (RPS).
+k6 is different from many other load testing tools in the way it handles hardware resources. A single k6 process will efficiently use all CPU cores on a load generator machine. Depending on the available resources, and taking into consideration some of the advice below, a single instance of k6 is able to run 30,000-40,000 simultaneous users (VUs). In some cases this number of VUs can generate up to 300,000 HTTP requests per second (RPS).
 
 Unless you need more than 100,000-300,000 requests per second (6-12M requests per minute), a single instance of k6 will likely be sufficient for your needs.
 
 Below we will explore what hardware and considerations are needed for generating different levels of load.
 
 ## OS fine-tuning
```

Comment:
As a casual, I feel uncomfortable running these commands.

These processes require on root and it looks like they modify kernal interaction and pre-defined user resource limits. As a reader, I think "why are these limits here? Am I doing something dangerous?" As a writer, I think, "Am I encouraging readers to copy and paste something without thinking about potentially serious ramifications?"

I looked at the "Fine Tuning OS" and see there's a bit more guidance. Still, though this section links that page, I'd suggest putting an admonition in this section.

```
<Blockquote mod="attention" title="">

The default limits protect your system from programs that might leak memory in huge quantities.
Consider increasing the limits only for the shell session that runs the test.

</Blockquote>

```


Maybe the language isn't correct, but something like that. Also, I see that the `ulimit` command by default doesn't persist, does the `sysctl` configuration change persist? If so, it might be good to provide details for reverting it.

Reply:
> As a casual, I feel uncomfortable running these commands.

Well, this article is directed more towards advanced k6 users, who are reaching the limits imposed by their OS. I agree that it should be as friendly as the rest of our documentation, but at some point we have to expect some familiarity from the reader.

We do explain the purpose of these commands below, and that they require root privileges (only the `sysctl` commands do, but we gloss over some of these subtleties that are better explained in the "Fine tuning OS" article). These are not "dangerous" by any means, but tweak the system to allow k6 (and any other program) to achieve maximum performance. These settings differ from OS and Linux distribution, but are usually set rather low by default, since the general public doesn't typically run programs that use tens of GBs of RAM, and opens thousands of concurrent TCP connections.

Also note that the settings changed by these commands are temporary, and will be reset to the system default after a reboot. If the user wants to make them permanent (presumably after they've confirmed the improvement), they can change some system files, which also require root privileges. The `ulimit` command applies only for the current shell session, so it has even less impact. This is all better explained in the "Fine tuning OS" article.

So I'm not sure what kind of warning would be helpful here, given the audience this article is aimed towards.

Reply:
> Well, this article is directed more towards advanced k6 users, 

Two things here: the article doesn't state directly who it's for (but the new intro does a much better job of setting expectations :+1:); how much does "advanced k6" use imply Linux knowledge?

I don't see what information with these extra two sentences.

```
These default limits are in place because ordinary programs don't need to open thousands of concurrent TCP connections or use dozens of GBs of RAM.
The `ulimit` changes apply only for the current shell, and the other settings reset when the system reboots.
For details, refer to [Fine-tuning OS](/misc/fine-tuning-os). 
```

Reply:
> the article doesn't state directly who it's for

None of our articles mention the intended audience AFAIK, but whoever is interested in "running large tests" and reaching the limits of their OS, would likely expect to read about advanced topics.

> how much does "advanced k6" use imply Linux knowledge?

I initially wrote "advanced user" in general, and still think it applies. I think there's some overlap between an advanced k6 user and an advanced computer user, so we don't need to explain every topic from scratch, especially since we already mention these details on the "Fine tuning OS" page.

That said, I expanded the intro a bit in 7e64d232. Let me know if this is an improvement.